SOPT_KakaoTalk/SOPT_KakaoTalk/Sources/ViewControllers/ProfileListViewController.swift

```diff
@@ -0,0 +1,138 @@
+        let label = UILabel()
+        label.text = "이름"
+        label.font = .systemFont(ofSize: 17, weight: .semibold)
+        
+        return label
+    }()
+
+    //MARK: - LifeCycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        self.layout()
+        self.config()
+        self.addTapGestures()
+        
+    }
+    
+    //MARK: - Properties
+    var name: String?
+    
+    
+    //MARK: - Functions
+    func dataBind() {
+        guard let name = self.name else {return}
+        self.nameLabel.text = name
+    }
+    
+    //MARK: - Private functions
+    private func addTapGestures() {
+        let tapGesture = UITapGestureRecognizer(target: self, action: #selector(touchupListView))
+        profileListView.addGestureRecognizer(tapGesture)
+    }
+    
+    private func presentProfileDetailVC() {
+        let detailVC = ProfileDetailViewController()
+        detailVC.modalPresentationStyle = .fullScreen
+        detailVC.name = self.nameLabel.text
+        detailVC.dataBind()
+        self.present(detailVC, animated: true, completion: nil)
+    }
+    
+    //MARK: - Objc functions
+    @objc func touchupListView() {
+        self.presentProfileDetailVC()
+    }
+
+}
+
+//MARK: - Extensions
+extension ProfileListViewController {
+    
+    //MARK: - Layout
+    private func layout() {
+        view.addSubviews([titleView, profileListView])
+        titleView.addSubviews([friendLabel, settingButton])
+        profileListView.addSubviews([profileImageView, nameLabel])
+        
+        //화면 전환 고려하여 Navigation bar로 수정해 보기
+        titleView.snp.makeConstraints { make in
+            make.top.equalTo(self.view.safeAreaLayoutGuide)
+            make.leading.trailing.equalToSuperview()
+            make.height.equalTo(52)
+        }
+        
+        friendLabel.snp.makeConstraints { make in
+            make.leading.equalToSuperview().offset(14)
+            make.centerY.equalToSuperview()
+        }
+        
+        settingButton.snp.makeConstraints { make in
+            make.leading.equalTo(self.friendLabel.snp.trailing).offset(2)
+            make.centerY.equalTo(self.friendLabel)
+        }
+        
+        profileListView.snp.makeConstraints { make in
+            make.top.equalTo(self.titleView.snp.bottom)
+            make.leading.trailing.equalToSuperview()
+            make.height.equalTo(70)
+        }
+        
+        profileImageView.snp.makeConstraints { make in
+            make.centerY.equalToSuperview()
+            make.leading.equalToSuperview().offset(14)
+            make.width.equalTo(59)
+            make.height.equalTo(58)
+        }
+
+        nameLabel.snp.makeConstraints { make in
+            make.centerY.equalToSuperview()
+            make.leading.equalTo(self.profileImageView.snp.trailing).offset(10)
+        }
+        
+    }
+    
+    //MARK: - Config
+    private func config() {
+        view.backgroundColor = .white
+
+    }
```

Comment:
PRPoint로 남겨주신 부분과 관련해서 (+ 그리고 저의 개인적인 코드 취향을 반영해서) 몇가지 리뷰를 남기자면 !!

저는 보통 config메서드 안에서도 어떤 UIComponent를 관리하는가?로 메서드를 다시 나눕니다.
예를 들어서 이 화면에서는 화면이 그려지는 시점에, 즉 초기화 되는 시점에서 label, image view 등의 초기화 작업이 한번에 이뤄지기 때문에 config 메서드 안에서 뷰의 배경색 정도를 지정할 수 있겠지만 

🤔 만약 서버에서 정보를 받아온 UI Componet의 변화가 필요하다면?
🤔 초기화 시점에서 모든 작업을 하는 것이 아니라면?
🤔 협업 시의 코드 스타일이 위에서 모든 초기화 작업을 하는 것 아니라면?

그렇다면 아래와 같이 코드를 수정해서 config 메서드를 관리할 수 있어요. ⬇️

```suggestion
    //MARK: - Config
    private func config() {
        view.backgroundColor = .white
        
        configLabel()
        configImageView()
    }
    
    private func configLabel() {
        [friendLabel, nameLabel].forEach {
            textColor = .black 
        }
    }
    
    private func configImageView() {
        if (특정 상황일 때) { 
            profileImageView.image = nil
        }
    }
```

위의 코드는 예시일뿐 .. 계속해서 어떤 코드/메서드로 관리하는 것이 좋을까? 고민해보는 것은 필요한 것 같아용
가독성 측면을 고려할 수도 있을 것 같고 .. !! 
코드를 계속 분리하다보면 비슷한 부분을 묶어서 base로 관리할 수도 있겠죠? 그러면 ViewController이 massive 해지는 것도 방지할 수 있을겁니당 

제가 너무 주저리 적어서 이해가 되지 않는다면 다시 답 남겨주세요를레히~

Reply:
당연히 기억합니다~!!!!!!!!!!!!
여러가지 많이 배워가요
저두 코드를 더 많이 써보면서 어떻게 관리하면 좋을지 계속해서 고민해 보겠습니다,, 😆
참고로 미니마니모 진짜 고등학교 다닐 때 친구들이 불렀던 별명이람니다,,, 😆